cli/medperf/commands/compatibility_test/utils.py

```diff
@@ -138,23 +138,23 @@ def create_test_dataset(
     # TODO: existing dataset could make problems
     # make some changes since this is a test dataset
     config.tmp_paths.remove(data_creation.dataset.path)
-    data_creation.dataset.write()
     if skip_data_preparation_step:
         data_creation.make_dataset_prepared()
     dataset = data_creation.dataset
+    old_generated_uid = dataset.generated_uid
+    old_path = dataset.path
 
     # prepare/check dataset
     DataPreparation.run(dataset.generated_uid)
 
     # update dataset generated_uid
-    old_path = dataset.path
-    generated_uid = get_folders_hash([dataset.data_path, dataset.labels_path])
-    dataset.generated_uid = generated_uid
-    dataset.write()
-    if dataset.input_data_hash != dataset.generated_uid:
```

Comment:
It seems previously  old uid was stored in `input_data_hash` but we do not update it. Why?

Reply:
`input_data_hash` is meant to be the hash of the raw data, which we know it doesn't change after preparation in this code snippet.
`generated_uid` (aka `output_data_hash`) is set to `input_data_hash` before preparation just because "technically the data is not prepared so rawdata == outputdata", but that is not really critical